internal/cache/devices_test.go

```diff
@@ -0,0 +1,153 @@
+package cache
+
+import (
+	"context"
+	"testing"
+
+	"github.com/edgexfoundry/device-sdk-go/internal/common"
+	"github.com/edgexfoundry/device-sdk-go/internal/mock"
+	"github.com/edgexfoundry/go-mod-core-contracts/models"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+)
+
+var ds []models.Device
+
+func init() {
+	common.ServiceName = "device-cache-test"
+	common.DeviceClient = &mock.DeviceClientMock{}
+	ctx := context.WithValue(context.Background(), common.CorrelationHeader, uuid.New().String())
+	ds, _ = common.DeviceClient.DevicesForServiceByName(common.ServiceName, ctx)
+}
+
+func TestNewDeviceCache(t *testing.T) {
+	dc := newDeviceCache([]models.Device{})
+	if _, ok := dc.(DeviceCache); !ok {
+		t.Error("the newDeviceCache function supposed to return a value which holds the DeviceCache type")
+	}
```

Comment:
This doesn't really test anything. It just verifies we can instantiate the deviceCache type, which is more of a Go runtime function. Unit tests should test business logic independent of infrastructure concerns, and the runtime is definitely a runtime concern. It's like trying to test the Earth.